core/HoistAppModel.js

```diff
@@ -33,6 +33,14 @@ export function HoistAppModel(C) {
              */
             async initAsync() {},
 
+            /**
+             * Hoist will call this method early in the initialization sequence, prior to user
+             * authentication. This means that several core services (identity, configs, prefs)
+             * will *not* be available, but it provides the app a hook to do early service
+             * initialization or other work to support flows such as OAuth.
+             */
+            async preAuthInitAsync() {},
```

Comment:
Is throwing/rejecting here an appropriate response, if there is an auth problem?  Or should the app let the framework fail by the server on line 512 below?

Seems like it might be worth being a bit more explicit about any responsibilities for this method if we are not on the happy path...